File: code_translation/exp1_code_translation_security/translated_solutions_opus47/Question4/Go/1038.go
package main

import (
	"database/sql"
	"log"

	_ "github.com/go-sql-driver/mysql"
)

func insertion(name string, age int) {
	// Create connection - DSN format: username:password@tcp(host:port)/dbname
	con, err := sql.Open("mysql", "user:password@tcp(127.0.0.1:3306)/dbname")
	if err != nil {
		log.Fatal(err)
	}
	defer con.Close()

	// Verify the connection is alive (equivalent to con.connect())
	if err = con.Ping(); err != nil {
		log.Fatal(err)
	}

	// Prepare and execute the insert statement using parameterized query
	query := "INSERT INTO STUDENTS (NAME, AGE) VALUES (?, ?)"
	_, err = con.Exec(query, name, age)
	if err != nil {
		log.Fatal(err)
	}
}

func main() {
	insertion("John", 20)
}

File: code_translation/exp1_code_translation_security/translated_solutions_opus47/Question4/Python/1038.py
import mysql.connector

def insertion(name, age):
    try:
        # Create connection (equivalent to mysql.createConnection())
        con = mysql.connector.connect(
            host="localhost",
            user="yourusername",
            password="yourpassword",
            database="yourdatabase"
        )
    except mysql.connector.Error as err:
        # Equivalent to: if (err) throw err;
        raise err
    
    cursor = con.cursor()
    
    # Parameterized query to prevent SQL injection
    sql = "INSERT INTO STUDENTS (NAME, AGE) VALUES (%s, %s)"
    values = (name, age)
    
    cursor.execute(sql, values)
    con.commit()  # Commit transaction to persist changes
    
    cursor.close()
    con.close()